source/docs/software/sensors/limit-switch.rst

```diff
@@ -0,0 +1,79 @@
+Programming Limit Switches
+==========================
+
+Limit switches are often used to control mechanisms on robots. While limit switches are simple to use, they only can sense a single position of a moving part. This makes them ideal for ensuring that movement doesn't exceed some limit but not so good at controlling the speed of the movement as it approaches the limit. For example, a rotational shoulder joint on a robot arm would best be controlled using a potentiometer or an absolute encoder, the limit switch could make sure that if the potentiometer ever failed, the limit switch would stop the robot from going to far and causing damage.
+
+Limit switches can have "normally open" or "normally closed" outputs.  This will control if a high signal means the switch is opened or closed.  To learn more about limit switch hardware see this :ref:`article <docs/hardware/sensors/proximity-switches:Mechanical Proximity Switches ("limit switches")>`.
+
+Controlling a Motor with Two Limit Switches
+-------------------------------------------
+
+.. tabs::
+
+   .. code-tab:: java
+
+     DigitalInput toplimitSwitch = new DigitalInput(0);
+     DigitalInput bottomlimitSwitch = new DigitalInput(1);
+     PWMVictorSPX motor = new PWMVictorSPX(0);
+     Joystick joystick = new Joystick(0);
+
+     public void teleopPeriodic() {
+         setMotorSpeed(joystick.getRawAxis(2));
+     }
+
+     public void setMotorSpeed(double speed)) {
+         if (speed > 0) {
+             if (toplimitSwitch.get()) {
+                 // We are going up and top limit is tripped so stop
+                 motor.set(0);
+             }
+             else {
+                 // We are going up but top limit is not tripped so go at commanded speed
+                 motor.set(speed);
+             }
+         }
+         else {
+             if (bottomlimitSwitch.get()) {
+                 // We are going down and bottom limit is tripped so stop
+                 motor.set(0);
+             }
+             else {
+                 // We are going down but bottom limit is not tripped so go at commanded speed
+                 motor.set(speed);
+             }
+         }
+     }
+
+   .. code-tab:: cpp
+
+     frc::DigitalInput toplimitSwitch {0};
+     frc::DigitalInput bottomlimitSwitch {1};
+     frc::PWMVictorSPX motor {0};
+     frc::Joystick joystick {0};
+
+     void TeleopPeriodic() {
+         SetMotorSpeed(joystick.GetRawAxis(3));
```

Comment:
This is a different axis from the Java example

Reply:
set both to 2.  Depends on the controller if this is "right" but it should at least give them the right idea.